api/src/main/java/io/strimzi/api/kafka/model/KafkaAuthorizationOpa.java

```diff
@@ -0,0 +1,115 @@
+import io.strimzi.crdgenerator.annotations.Example;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+/**
+ * Configures the broker authorization to use Open Policy Agent as an authorization and policy server.
+ */
+@DescriptionFile
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "url", "allowOnError", "initialCacheCapacity", "maximumCacheSize", "expireAfterMs", "token", "superUsers"})
+@EqualsAndHashCode
+public class KafkaAuthorizationOpa extends KafkaAuthorization {
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_OPA = "opa";
+
+    public static final String AUTHORIZER_CLASS_NAME = "com.bisnode.kafka.authorization.OpaAuthorizer";
+
+    private List<String> superUsers;
+    private String url;
+    private boolean allowOnError = false;
+    private int initialCacheCapacity = 5000;
+    private int maximumCacheSize = 50000;
+    private long expireAfterMs = 3600000;
+
+    @Description("Must be `" + TYPE_OPA + "`")
+    @Override
+    public String getType() {
+        return TYPE_OPA;
+    }
+
+    @Description("List of super users. Should contain list of user principals which should get unlimited access rights.")
+    @Example("- CN=my-user\n" +
+             "- CN=my-other-user")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public List<String> getSuperUsers() {
+        return superUsers;
+    }
+
+    public void setSuperUsers(List<String> superUsers) {
+        this.superUsers = superUsers;
+    }
+
+    @Description("URL of the OPA server with the policy name which should be queried. " +
+            "This option is required.")
+    @Example("http://opa:8181/v1/data/kafka/authz/allow")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    @Description("Defines the authorizer behaviour when the OPA calls fail. " +
+            "Defaults to `false`.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public boolean isAllowOnError() {
+        return allowOnError;
+    }
+
+    public void setAllowOnError(boolean allowOnError) {
+        this.allowOnError = allowOnError;
+    }
+
+    @Description("Initial capacity of the decision cache. " +
+            "Defaults to `5000`.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public int getInitialCacheCapacity() {
+        return initialCacheCapacity;
+    }
+
+    public void setInitialCacheCapacity(int initialCacheCapacity) {
+        this.initialCacheCapacity = initialCacheCapacity;
+    }
+
+    @Description("Maximum capacity of the decision cache. " +
+            "Defaults to `50000`.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public int getMaximumCacheSize() {
```

Comment:
is there a way for disabling cache? Setting max cache size to 0?

Reply:
I'm not sure setting it to 0 would work. I do not think disabling the cache makes really any sense given how Apache Kafka works. The ACLs are evaluated every single time. So for example without the cache, every single fetch or produce request would mean HTTP call to OPA. So disabling the cache would criple the performance on every level. The reason for these cache options is mainly to manage the cost when the cache is growing and balance the memeory consumption.